search-service/src/main/kotlin/com/egm/stellio/search/service/AttributeInstanceService.kt

```diff
@@ -309,6 +318,130 @@ class AttributeInstanceService(
         )
     }
 
+    private suspend fun retrieveTeaUUIDAndPayload(
+        entityId: URI,
+        attributeName: ExpandedTerm,
+        instanceId: URI
+    ): Either<APIException, Pair<UUID, Map<String, List<Any>>>> {
+        val modifiedQuery =
+            """
+            SELECT temporal_entity_attribute, payload
+            FROM attribute_instance
+            WHERE temporal_entity_attribute = ( 
+                SELECT id 
+                FROM temporal_entity_attribute 
+                WHERE entity_id = :entity_id 
+                AND attribute_name = :attribute_name
+            )
+            AND instance_id = :instance_id
+            """.trimIndent()
+
+        return databaseClient
+            .sql(modifiedQuery)
+            .bind("entity_id", entityId)
+            .bind("attribute_name", attributeName)
+            .bind("instance_id", instanceId)
+            .oneToResult(
+                ResourceNotFoundException(
+                    attributeOrInstanceNotFoundMessage(attributeName, instanceId.toString())
+                )
+            ) {
+                Pair(
+                    it["temporal_entity_attribute"] as UUID,
+                    JsonUtils.deserializeObject(toJsonString(it["payload"])) as Map<String, List<Any>>
+                )
+            }
+    }
+
+    @Transactional
+    suspend fun modifyAttributeInstance(
+        entityId: URI,
+        attributeName: ExpandedTerm,
+        instanceId: URI,
+        expandedPayload: Map<String, List<Any>>
+    ): Either<APIException, Unit> = either {
+        val (teaUUID, oldPayload) = retrieveTeaUUIDAndPayload(entityId, attributeName, instanceId).bind()
+        val payload = expandedPayload.toMutableMap()
+        oldPayload.forEach {
+            payload.putIfAbsent(it.key, it.value)
+        }
+        val measuredValue = valueToDoubleOrNull(
+            (expandedPayload[NGSILD_PROPERTY_VALUE] as List<Map<String, String>>).first()[JSONLD_VALUE_KW]!!
+        )!!
+        val newInstance = AttributeInstance(
+            time = getPropertyValueFromMapAsDateTime(expandedPayload, NGSILD_OBSERVED_AT_PROPERTY)!!,
+            measuredValue = measuredValue,
+            payload = payload,
+            instanceId = instanceId,
+            temporalEntityAttribute = teaUUID,
+            timeProperty = AttributeInstance.TemporalProperty.OBSERVED_AT
+
+        )
+        deleteInstance(entityId, attributeName, instanceId).bind()
+        create(newInstance).bind()
```

Comment:
not sure if we should update the existing instance (keeping the current `instanceId` - my bet) or it should be delete / create